proseco/acq.py

```diff
@@ -27,6 +33,86 @@
 )
 
 
+def load_maxmags() -> dict:
+    """
+    Load maxmags from disk.
+
+    Maxmag data made with analysis/interpolate-maxmags-contour-plot.ipynb
+
+    """
+    path = Path(__file__).parent / "maxmags.npz"
+    with np.load(path) as arrs:
+        out = dict(arrs)
+    for box_sz in out["box_sizes"]:
+        out[box_sz] = out.pop(f"vals_{box_sz}")
+    return out
+
+
+MAXMAGS = LazyDict(load_maxmags)
+
+
+def get_maxmag(box_size: int, t_ccd: float) -> float:
+    """
+    Get maxmag for given box_size and t_ccd.
+
+    This corresponds to the MAXMAG that results in exactly 50 search hits. See
+    https://occweb.cfa.harvard.edu/twiki/bin/view/Aspect/PeaMaxMagTesting.
+
+    Maxmag data made with analysis/interpolate-maxmags-contour-plot.ipynb.
+
+    :param box_size: box size (int, arcsec)
+    :param t_ccd: CCD temperature (float, C)
+    :returns: maxmag (float)
+    """
+    if t_ccd < -10.0:
+        return 11.2
+    if t_ccd > 0:
+        warnings.warn(f"Clipping {t_ccd=} to 0.0 for interpolating MAXMAGs table")
+        t_ccd = 0.0
+
+    maxmag = np.interp(t_ccd, xp=MAXMAGS["t_ccds"], fp=MAXMAGS[box_size])
+
+    # Need to round to nearest 0.01 mag because of values that are just slightly
+    # below 11.2. These need to become exactly 11.2.
+    return maxmag.round(2)
+
+
+def filter_box_sizes_for_maxmag(
+    mag: float, mag_err: float, box_sizes: np.ndarray, t_ccd: float
+) -> np.ndarray:
+    """Filter the list of box sizes
+
+    First compute the smallest allowed value of MAXMAG for this star, which is
+    the star mag + 3 times the star mag error (clipped to be within 0.5 to 1.5
+    mag, nominal).
+
+    For each box size and t_ccd compute the MAXMAG that keeps the search hits
+    at exactly 50. Then keep the box sizes where MAXMAG is less than the star
+    minimum MAXMAG.
+
+    :param mag: star mag (float)
+    :param mag_err: star mag error (float)
+    :param box_sizes: ndarray of box sizes (float, arcsec)
+    :param t_ccd: CCD temperature (float, C)
+    :returns: ndarray of box sizes (float, arcsec)
+    """
+    maxmag_min = mag + np.clip(3 * mag_err, ACA.min_delta_maxmag, ACA.max_delta_maxmag)
+
+    # Hard limit of ACA.max_maxmag (11.2) from operational change made in 2019.
+    # We always accept a maxmax of 11.2 regardless of star mag / mag_err.
+    maxmag_min = maxmag_min.clip(None, ACA.max_maxmag)
+
+    ok = [maxmag_min <= get_maxmag(box_size, t_ccd) for box_size in box_sizes]
+    out: np.ndarray = box_sizes[ok]
+
+    # Always allow at least the smallest box size. This situation will be
```

Comment:
Why do we need to allow at least the smallest box size and how is this flagged?

Reply:
Because it will be better to have an acq star with a 60 arcsec box (where the actual acq prob is less than the model) than nothing at all. It will be flagged in sparkles which will check the commanded maxmag vs allowed given T_ccd and box size.

Reply:
Sounds fine but where should we track the lien?

Reply:
Sparkles issue.